torch/backends/_nnapi/serializer.py

```diff
@@ -1288,6 +1298,15 @@ def add_upsample_nearest2d(self, node):
 
         out_shape = (image_oper.shape[0], image_oper.shape[1], out_h, out_w)
         use_nchw = image_oper.use_nchw()
+        out_id = self.add_tensor_operand(node.outputsAt(0), image_oper._replace(shape=out_shape))
+
+        # Handle variable input size
+        for dim in (2, 3):   # h, w indices
```

Comment:
Dimensions 0 and 1 are not handled.  We should either forward them (more complete coverage) or assert that they are fixed (probably fine in practice).

Reply:
Raising exceptions in those cases for now, will add support in a separate PR for both conv and nearest sample